kai/lib.go

```diff
@@ -79,29 +76,51 @@ func GetInventoryReport(cfg *config.Application) (inventory.Report, error) {
 	ch := channels{
 		reportItem: make(chan inventory.ReportItem),
 		errors:     make(chan error),
+		stopper:    make(chan struct{}, 1),
 	}
-	nsCh := make(chan k8sNamespace)
 
-	go fetchNamespaces(kubeconfig, cfg, nsCh)
+	namespaces, err := fetchNamespaces(kubeconfig, cfg)
+	if err != nil {
+		return inventory.Report{}, err
+	}
 
-	total := 0
-	for ns := range nsCh {
-		if ns.Err != nil {
-			return inventory.Report{}, fmt.Errorf("failed to resolve namespace: %w", ns.Err)
-		}
+	// fill the queue of namespaces to process
+	queue := make(chan string, len(namespaces))
+	for _, n := range namespaces {
+		queue <- n
+	}
+	close(queue)
 
-		// Does a "get pods" for the specified namespace and returns the unique set of images to the ch.reportItem channel
-		go fetchPodsInNamespace(kubeconfig, cfg.Kubernetes, ns.Name, ch)
-		total++
+	// get pods from namespaces using a worker pool pattern
+	for i := int64(0); i < cfg.Kubernetes.WorkerPoolSize; i++ {
```

Comment:
Just out of curiosity, why are we explicitly making `i` an `int64` here? Is it in case the worker pool size becomes extremely large?

Reply:
Updated @subecho appreciate the curiosity! 🥇